models/stage.model.js

```diff
@@ -0,0 +1,32 @@
+import mongoose from "mongoose";
+
+const { Schema } = mongoose;
+
+const StageSchema = new Schema(
+  {
+    submissionType: {
+      type: [Schema.Types.ObjectId],
```

Comment:
I think there should be one submission to one stage. If so, remove the square braces from here.